pyanaconda/payload/__init__.py

```diff
@@ -989,9 +988,30 @@ def __init__(self, data):
         # environment.
         self._environmentAddons = {}
 
+    def setup(self, storage, instClass):
+        """Do any PackagePayload-specific setup."""
+        super(PackagePayload, self).setup(storage, instClass)
+
+        # Add any known external repos from the instClass
+        #  Activate if requested by instClass
+        for external_repo in instClass.external_addon_repos:
+            if isinstance(external_repo, self.data.RepoData):
+                log.debug("adding external_repo: name=%s enabled=%s",
+                          external_repo.name, external_repo.enabled)
+                self.addRepo(external_repo, enabled=external_repo.enabled)
+            else:
+                log.debug("external_addon_repo '%s' not of type '%s' ignored",
+                          external_repo, self.data.RepoData)
+
     def preInstall(self):
         super(PackagePayload, self).preInstall()
 
+        # remove any disabled repos
+        for repo in self.data.repo.dataList():
+            if not self.isRepoEnabled(repo.name):
+                log.debug("Removing disabled repo %s", repo.name)
+                self.removeRepo(repo.name)
+
```

Comment:
Why we need to remove disabled repos? They should be ignored for installation anyway.

Reply:
They are ignored by the installer, but pykickstart sees them in the list and adds them to anaconda-ks.cfg (even though they are disabled)